examples/cuda/explicit_cuda_stream.cu

```diff
@@ -0,0 +1,50 @@
+#include <thrust/device_vector.h>
+#include <thrust/execution_policy.h> // For thrust::device
+#include <thrust/reduce.h>
+#include <thrust/sequence.h>
+
+#include <cuda_runtime.h>
+
+#include <iostream>
+
+// This example shows how to execute a Thrust device algorithm on an explicit
+// CUDA stream. The simple program below fills a vector with the numbers
+// [0, 1000) (thrust::sequence) and then sums them (thrust::reduce), executing
+// both algorithms on the same custom CUDA stream.
```

Comment:
It feels like after https://github.com/NVIDIA/thrust/pull/1568 we should have a description of the synchronization before and introduce the `no_sync` version as well. 

```suggestion
// both algorithms on the same custom CUDA stream using the execution policy. 
// Thrust provides two execution policies that accept CUDA streams that differ in when/if they synchronize the stream:
// 1. thrust::cuda::par.on(stream)
//      - `stream` will *always* be synchronized before an algorithm returns
// 2. thrust::cuda::par_nosync.on(stream)
//      - `stream` will only be synchronized when necessary for correctness (e.g., returning a result from `thrust::reduce`)
```